darts/timeseries.py

```diff
@@ -1164,41 +1515,51 @@ def resample(self, freq: str, method: str = 'pad') -> 'TimeSeries':
             A reindexed TimeSeries with given frequency.
         """
 
-        new_df = self.pd_dataframe().asfreq(freq, method=method)
+        resample = self._xa.resample({self._time_dim: freq})
 
-        return TimeSeries(new_df, freq)
+        # TODO: check
+        if method == 'pad':
+            new_xa = resample.pad()
+        elif method == 'bfill':
+            new_xa = resample.backfill()
+        else:
+            raise_log(ValueError('Unknown method: {}'.format(method)), logger)
+        return TimeSeries(new_xa)
 
-    def is_within_range(self,
-                        ts: pd.Timestamp) -> bool:
+    def is_within_range(self, ts: Union[pd.Timestamp, int]) -> bool:
         """
-        Check whether a given timestamp is withing the time interval of this time series
+        Check whether a given timestamp or integer is withing the time interval of this time series.
+        If a timestamp is provided, it does not need to be *in* the time series.
 
         Parameters
         ----------
         ts
-            The `pandas.Timestamp` to check
+            The `pandas.Timestamp` or integer to check
```

Comment:
For a typical user it might sometimes be confusing if they read all the time of integers when we do something with time indexes (probably most users provide there own time index). Maybe we can specify:
The `pandas.Timestamp` (if indexed with DatetimeIndex) or integer (if indexed with RangeIndex) to check.